onnxruntime/core/optimizer/initializer.cc

```diff
@@ -0,0 +1,54 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include "core/optimizer/initializer.h"
+
+#include "gsl/gsl"
+
+#include "core/common/path.h"
+#include "core/framework/tensorprotoutils.h"
+#include "core/framework/tensor_external_data_info.h"
+#include "core/platform/env.h"
+
+namespace onnxruntime {
+Status Initializer::ReadExternalRawData(
+    const ONNX_NAMESPACE::TensorProto& tensor_proto, const Path& model_path, std::vector<char>& raw_data) {
+  ORT_RETURN_IF_NOT(
+      tensor_proto.data_type() != ONNX_NAMESPACE::TensorProto_DataType_UNDEFINED &&
+          tensor_proto.data_type() != ONNX_NAMESPACE::TensorProto_DataType_STRING,
+      "External data type must not be UNDEFINED or STRING.");
+
+  ORT_RETURN_IF(model_path.IsEmpty(), "model_path must not be empty.");
+
```

Comment:
Might be good to add a bit of info to the error message on why it might be empty, as it's fairly indirect to get from this message to how the model was created/loaded